.github/workflows/docs.yml

```diff
@@ -19,8 +28,18 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout
+      # We have two checkout steps running based on
+      # whether the workflow was manually trigerred.
+      # If manually trigerred, we use the provided ref,
+      # otherwise we use the default ref.
+      - name: Checkout (on release)
+        if:  github.event_name != 'workflow_dispatch'
+        uses: actions/checkout@v3
+      - name: Checkout (manual)
+        if:  github.event_name == 'workflow_dispatch'
         uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.head }}
```

Comment:
I believe you can collapse these into one and do:

```
ref: ${{ inputs.head || github.ref }}
```

But I'm not certain about it, so I opted for the duplicated version for now.